src/widget/radio.rs

```diff
@@ -0,0 +1,148 @@
+    ) -> impl Widget<T> {
+        let mut col = Column::new();
+        for (label, variant) in variants.into_iter() {
+            let radio = Radio::new(label, variant);
+            col.add_child(Padding::uniform(5.0, radio), 0.0);
+        }
+        col
+    }
+}
+
+/// A single radio button
+#[derive(Debug, Clone)]
+pub struct Radio<T: Data + PartialEq + 'static> {
+    phantom: PhantomData<T>,
+}
+
+impl<T: Data + PartialEq + 'static> Radio<T> {
+    pub fn new(label: impl Into<LabelText<T>>, variant: T) -> impl Widget<T> {
+        let mut row = Row::new();
+        row.add_child(RadioRaw { variant }, 0.0);
+        row.add_child(Label::new(label), 1.0);
+        Align::vertical(UnitPoint::CENTER, row)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct RadioRaw<T: Data + PartialEq> {
+    variant: T,
+}
+
+impl<T: Data + PartialEq> RadioRaw<T> {}
+
+impl<T: Data + PartialEq> Widget<T> for RadioRaw<T> {
+    fn paint(&mut self, paint_ctx: &mut PaintCtx, base_state: &BaseState, data: &T, env: &Env) {
+        let size = env.get(theme::BASIC_WIDGET_HEIGHT);
+
+        let circle = Circle::new((size / 2., size / 2.), 7.);
+
+        //Paint the background
+        let background_gradient = LinearGradient::new(
+            UnitPoint::TOP,
+            UnitPoint::BOTTOM,
+            (
+                env.get(theme::BACKGROUND_LIGHT),
+                env.get(theme::BACKGROUND_DARK),
+            ),
+        );
+
+        paint_ctx.fill(circle, &background_gradient);
+
+        let border_color = if base_state.is_hot() {
+            env.get(theme::BORDER_LIGHT)
+        } else {
+            env.get(theme::BORDER)
+        };
+
+        paint_ctx.stroke(circle, &border_color, 1.);
+
+        // If data enum matches our variant
+        if *data == self.variant {
+            let inner_circle = Circle::new((size / 2., size / 2.), 2.);
+
+            paint_ctx.fill(inner_circle, &env.get(theme::LABEL_COLOR));
+        }
+    }
+
+    fn layout(
+        &mut self,
+        _layout_ctx: &mut LayoutCtx,
+        bc: &BoxConstraints,
+        _data: &T,
+        env: &Env,
+    ) -> Size {
+        bc.constrain(Size::new(
+            env.get(theme::BASIC_WIDGET_HEIGHT),
+            env.get(theme::BASIC_WIDGET_HEIGHT),
+        ))
+    }
+
+    fn event(
+        &mut self,
+        event: &Event,
+        ctx: &mut EventCtx,
+        data: &mut T,
+        _env: &Env,
+    ) -> Option<Action> {
+        match event {
+            Event::MouseDown(_) => {
+                ctx.set_active(true);
+                ctx.invalidate();
+            }
+            Event::MouseUp(_) => {
+                if ctx.is_active() {
+                    ctx.set_active(false);
+                    if ctx.is_hot() {
+                        *data = self.variant.clone();
+                    }
+                    ctx.invalidate();
+                }
+            }
+            Event::MouseMoved(_) => {
```

Comment:
Dang, I totally forgot about `HotChanged`. I'm gonna go through all the widgets and make sure I use that correctly and make a new PR for that. For now I just fixed Checkbox because it's identical to Radio in this regard.